rasa_core/policies/ensemble.py

```diff
@@ -216,27 +218,32 @@ def probabilities_using_best_policy(self, tracker, domain):
                 result = probabilities
                 best_policy_name = 'policy_{}_{}'.format(i, type(p).__name__)
 
-        policy_names = [type(p).__name__ for p in self.policies]
-
-        # Trigger the fallback policy when ActionListen is predicted after
-        # a user utterance. This is done on the condition that: a fallback
-        # policy is present, there was just a user message and the predicted
-        # action is action_listen by a policy other than the MemoizationPolicy
+        if (result.index(max_confidence) ==
```

Comment:
I think it makes more sense to check if a fallback policy is present first, because otherwise we don't need to check for all this

Reply:
I'm not sure here. Then we need to iterate through `policies` arrays to find fallback policy every time. And I think, this way it is easier for users to override this behavior

Reply:
haha since we decided to make it specific to the fallback policy, it is fallback policy specific 😄

Reply:
I think either way is fine :+1:

Reply:
to me, if I were a user and I wanted to create specific policy for this, this would be easier to change

Reply:
🤷‍♀️ i still think it makes more sense the other way, but whatever, this PR needs to be merged so lets merge it

Reply:
dark side of the force clouds your vision, come to the light side 😂